mcts.py

```diff
@@ -147,7 +149,10 @@ def select_leaf(self):
                 current = current.maybe_add_child(pass_move)
                 continue
 
+            # NOTE(#379): child_action_score depends on current.N
```

Comment:
I don't see why we need this - why not just remove the initial current.N += 1 at the top of the while loop?

Reply:
that doesn't work when the node hasn't been expanded, though, correct?  We don't detect that we picked it if we never incremented N.

Reply:
`node.is_expanded` doesn't refer to `node.N`, so I'm not sure what you mean.

Reply:
it means N is not incremented in the situation where virtual losses are not sufficient to prevent the same node from being selected.